app/assets/javascripts/settings/ldap-conn-test.js

```diff
@@ -0,0 +1,68 @@
+$(function () {
+  function NoneEmpty (arr) {
+    for (var i = 0; i < arr.length; i++) {
+      if (arr[i] === '') return false
+    }
+    return true
+  };
+
+  $(document).ready(function () {
+    // Elements specific to LDAP Connector page; check for existence before changing
+    if ($('#ldap_conn_save').length > 0 && $('#ldap_conn_message').length > 0) {
+      $('#ldap_conn_save').prop('disabled', 'disabled')
+      $('#ldap_conn_message')[0].style.color = 'black'
+    };
+
+    // Run LDAP Connection Test with button click
+    $('#ldap_conn_test').click(function (event) {
+      // Get values from form
+      var host = $('#dex_connector_ldap_host').val()
+      var port = $('#dex_connector_ldap_port').val()
+      var startTLS = $('#dex_connector_ldap_start_tls_true').parent().hasClass('active')
+      var cert = $('#dex_connector_ldap_certificate').val()
+      // Flip logic of #anonymous_bind_toggle since expanded div means anonymous binding is 'false'
+      var anonBind = ($('#anonymous_bind_toggle').attr('aria-expanded') === 'false')
+      var dn = $('#dex_connector_ldap_bind_dn').val()
+      var pass = $('#dex_connector_ldap_bind_pw').val()
+      var baseDN = $('#dex_connector_ldap_user_base_dn').val()
+      var filter = $('#dex_connector_ldap_user_filter').val()
+
+      var baseURL = '/settings/ldap_test'
+      var valMessage = ''
+
+      var paramArray = [host, port, cert, dn, pass, baseDN, filter]
+      var allParamsEntered = NoneEmpty(paramArray)
+
+      // Check if form inputs are filled in
+      if (allParamsEntered) {
+        $.post(baseURL, {
+          host: host,
+          port: port,
+          start_tls: startTLS,
+          cert: cert,
+          anon_bind: anonBind,
+          dn: dn,
+          pass: pass,
+          base_dn: baseDN,
+          filter: filter
+        }).done(function (data) {
+          if (data.result.test_pass) {
+            valMessage = 'Sucessfully validated connection to LDAP server'
+            $('#ldap_conn_save').prop('disabled', null)
+            $('#ldap_conn_message')[0].style.color = '#04a36f'
```

Comment:
Instead of manipulating the color directly in the element, it would be nicer to use classes to map the state of the element. You could add a class like `valid` or `#ldap_conn_message--valid` and add it here. Same applies for the failure/invalid message below.

Reply:
Fixed now.

Reply:
Added classes:  ldap-conn-message-default, ldap-conn-message-invalid, ldap-conn-message-valid